psalm-baseline.xml

```diff
@@ -1,11 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<files psalm-version="4.x-dev@">
+<files psalm-version="5.x-dev@">
+  <file src="src/AbstractCollection.php">
+    <MissingTemplateParam occurrences="1">
+      <code>AbstractCollection</code>
+    </MissingTemplateParam>
+  </file>
+  <file src="src/AuthenticationService.php">
+    <LessSpecificReturnStatement occurrences="2">
+      <code>$this-&gt;authenticators()-&gt;load($name, $config)</code>
+      <code>$this-&gt;identifiers()-&gt;load($name, $config)</code>
+    </LessSpecificReturnStatement>
+    <MoreSpecificReturnType occurrences="2">
+      <code>\Authentication\Authenticator\AuthenticatorInterface</code>
+      <code>\Authentication\Identifier\IdentifierInterface</code>
+    </MoreSpecificReturnType>
+  </file>
   <file src="src/Authenticator/AuthenticatorCollection.php">
+    <ImplementedReturnTypeMismatch occurrences="1">
+      <code>\Authentication\Authenticator\AuthenticatorInterface|null</code>
+    </ImplementedReturnTypeMismatch>
     <MoreSpecificImplementedParamType occurrences="1">
       <code>$class</code>
     </MoreSpecificImplementedParamType>
   </file>
   <file src="src/Identifier/IdentifierCollection.php">
+    <ImplementedReturnTypeMismatch occurrences="1">
+      <code>\Authentication\Identifier\IdentifierInterface|null</code>
+    </ImplementedReturnTypeMismatch>
```

Comment:
What is causing the mismatch?

Reply:
The error is:
```
ERROR: ImplementedReturnTypeMismatch - src/Identifier/IdentifierCollection.php:25:12 - The inherited return type 'object' for Cake\Core\ObjectRegistry::get is different to the implemented return type for Cake\Core\ObjectRegistry::get 'Authentication\Identifier\IdentifierInterface|null' 
```
because `ObjectRegistry::get()` returns a generic object

Reply:
Adding proper annotation like [this](https://github.com/cakephp/cakephp/blob/5.x/src/Controller/ComponentRegistry.php#L30) will make static analyzers understand the specific type returned.

Reply:
Tried that but wasn't sucessfull because here we are using an `AbstractCollection` in between (and I am not able to find a combination with whom the static analysis gods are happy with)

Reply:
I'll see if I can please the stan gods.